subsys/bluetooth/host/att_internal.h

```diff
@@ -271,7 +271,7 @@ struct bt_att_signed_write_cmd {
 } __packed;
 
 typedef void (*bt_att_func_t)(struct bt_conn *conn, uint8_t err,
-			      const void *pdu, uint16_t length,
+			      const void *pdu, ssize_t length,
 			      void *user_data);
```

Comment:
Rather than using the `length` field for non-ATT errors and misusing `ssize_t` (see https://github.com/zephyrproject-rtos/zephyr/issues/44454), would it make sense to make the `err` here an `int` so that we can use positive values for ATT errors, and negative for errno?

Reply:
> Rather than using the `length` field for non-ATT errors and misusing `ssize_t` (see #44454), would it make sense to make the `err` here an `int` so that we can use positive values for ATT errors, and negative for errno?

Good idea. The reason I didn't do that is the convention that we have in Zephyr GATT that ATT Error codes returned from app are negative.

gatt.h:
```
/**
 *  @brief Construct error return value for attribute read and write callbacks.
 *
 *  @param _att_err ATT error code
 *
 *  @return Appropriate error code for the attribute callbacks.
 */
#define BT_GATT_ERR(_att_err) (-(_att_err))
```
att.c:
```
static uint8_t err_to_att(int err)
{
	LOG_DBG("%d", err);

	if (err < 0 && err >= -0xff) {
		return -err;
	}

	return BT_ATT_ERR_UNLIKELY;
}
```

I thought that it would be not clear why in this particular case we don't follow the schema.
To follow this convention I would need to change all the `func` call occurrences to pass negative ATT Error, and in gatt.c revert it back to positive value (awful solution).

But, if we agree that in this particular case we'll use positive ATT Error code and negative errno, I'll update the PR.

Reply:
I prefer this rather than "free-real-estate"-ing `length`, yeah